core/net/src/hybrid/service.rs

```diff
@@ -152,19 +139,38 @@ pub async fn start_network(
         }
     };
 
+    {
+        let client = client.clone();
+        typed::bind(
+            ya_core_model::net::local::BUS_ID,
+            move |_: ya_core_model::net::local::Shutdown| {
+                let mut client = client.clone();
+                async move {
+                    client
+                        .shutdown()
+                        .map_err(|e| GenericNetError(e.to_string()))
+                        .await
+                }
+            },
+        );
+    }
+
     bind_local_bus(
+        client.clone(),
```

Comment:
of course, but it would be a longer refactor because then, for example, the State::forward_sink signature would have to remove the clinet argument.the API will change, so there will be an opportunity to plow through this module.